File: android-client/app/src/main/java/com/example/awscognito/data/model/FeedItem.kt
package com.example.awscognito.data.model

data class FeedItem(
    val id: String,
    val title: String,
    val content: String,
    val type: String
) {
    companion object {
        const val TYPE_ANNOUNCEMENT = "announcement"
        const val TYPE_UPDATE = "update"
        const val TYPE_REPORT = "report"
    }
}

File: ios-client/AWSCognito/AWSCognito/Data/Models/FeedItem.swift
//
//  FeedItem.swift
//  AWSCognito
//

import Foundation
import SwiftUI

struct FeedItem: Codable, Identifiable {
    let id: String
    let title: String
    let content: String
    let type: String

    var typeColor: Color {
        switch type.lowercased() {
        case "announcement":
            return .blue
        case "update":
            return .green
        case "report":
            return .yellow
        default:
            return .gray
        }
    }
}
